### PR TITLE
refactor(connector): [Worldpay] Remove Default Case Handling

### LINKED ISSUE #2290: [REFACTOR]: [Worldpay] Remove Default Case Handling
### :memo: Feature Description

- We utilize match statements to make pivotal decisions, such as generating requests based on the payment method type and managing responses received from the connector. 
- These conditions generally go hand in hand with enum variants. 
- Default case is used because a match statement needs to be exhaustive i.e. every variant needs to be covered. 
- So, if all the explicit cases are handled then default is used to handle the rest.
- Each connector have these match statements but many of them don’t provide reference to each variant in their default case, rather a `_` is put to handle all the other cases. 
- This approach carries a risk because developers may inadvertently overlook the need for explicit handling of the new cases.

### :hammer: Possible Implementation

- Instead of relying on a default match case `_`, developers should handle each and every variant explicitly. 
- By doing so, if there are any changes in the future, they can readily assess the impact of their modifications simply by compiling the code.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1955

:bookmark: Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by sudharsangs**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Instead of relying on a default match case `_` I have added conditions for each type in match statements. 

Changes are made at https://github.com/juspay/hyperswitch/blob/main/crates/router/src/connector/worldpay/transformers.rs

## Motivation and Context
Closes #2290

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/worldpay/transformers.rs [hunk 1]
```diff
@@ -144,7 +144,24 @@ impl TryFrom<&types::ConnectorAuthType> for WorldpayAuthType {
                     api_key: Secret::new(auth_header),
                 })
             }
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
+            types::ConnectorAuthType::CurrencyAuthKey { auth_key_map: _ } => {
```
Comment (prasunna09) [inline crates/router/src/connector/worldpay/transformers.rs:147]: Hi @sudharsangs, 
Can you please revert back this change as wildcard match arms (_) is not in scope for AuthType?

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/worldpay/transformers.rs (hunk 1 above)
Comment (sudharsangs) [reply crates/router/src/connector/worldpay/transformers.rs]: I've made the change @prasunna09

**Review by prasunna09 — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]
Looks good to me!

Thanks for the PR, @sudharsangs!

**Comment by prasunna09** [comment]
@sudharsangs  Can you please revert back the Cargo.lock file?

**Comment by sudharsangs** [comment]
> Hey @sudharsangs , Please revert back the cargo.lock file. Run `git checkout main -- Cargo.lock` command and push your code.

I did a ` git checkout upstream/main -- Cargo.lock`   but it's not changing I am not sure why

**Review by Sangamesh26 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Comment by VedantKhairnar** [comment]
Hey @sudharsangs  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**